sql/expression/comparison.go

```diff
@@ -6,51 +6,74 @@ import (
 	"gopkg.in/src-d/go-mysql-server.v0/sql"
 )
 
-// Comparison is an expression that compares an expression against another.
-type Comparison struct {
+// Compararer implements a comparison expression.
+type Comparer interface {
+	sql.Expression
+	IsComparison() bool
+	Left() sql.Expression
+	Right() sql.Expression
+	SetLeft(sql.Expression)
+	SetRight(sql.Expression)
+}
+
+type comparison struct {
```

Comment:
this should be public, to be able to check if the expression is a Comparison on rules.

Reply:
That check is done by the interface `Comparer`, because `comparison` is just used as an embedded type in the real comparison expressions.

If you try a type assertion on a interface by the embedded type of the type, it doesn't work.
See: https://play.golang.org/p/NJHOB7G27mc